sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -1222,12 +1225,18 @@ class DatasetSuite extends QueryTest
     assert(result == Set(ClassData("a", 1) -> null, ClassData("b", 2) -> ClassData("x", 2)))
   }
 
-  test("better error message when use java reserved keyword as field name") {
+  test("better error message when use invalid java identifier as field name") {
     val e = intercept[UnsupportedOperationException] {
       Seq(InvalidInJava(1)).toDS()
     }
     assert(e.getMessage.contains(
-      "`abstract` is a reserved keyword and cannot be used as field name"))
+      "`abstract` is not a valid identifier of Java and cannot be used as field name"))
+
+    val e2 = intercept[UnsupportedOperationException] {
```

Comment:
Could you add this test in a new test block instead of adding the existing one? btw, it seems better to place these tests for `ScalaReflection` in `ScalaReflectionRelationSuite`.